pkg/controller/backup_sync_controller.go

```diff
@@ -20,58 +20,69 @@ import (
 	"context"
 	"time"
 
-	"github.com/golang/glog"
+	"github.com/sirupsen/logrus"
 
 	"k8s.io/apimachinery/pkg/api/errors"
 	"k8s.io/apimachinery/pkg/util/wait"
 
 	"github.com/heptio/ark/pkg/cloudprovider"
 	arkv1client "github.com/heptio/ark/pkg/generated/clientset/typed/ark/v1"
+	"github.com/heptio/ark/pkg/util/kube"
 )
 
 type backupSyncController struct {
 	client        arkv1client.BackupsGetter
 	backupService cloudprovider.BackupService
 	bucket        string
 	syncPeriod    time.Duration
+	logger        *logrus.Logger
 }
 
-func NewBackupSyncController(client arkv1client.BackupsGetter, backupService cloudprovider.BackupService, bucket string, syncPeriod time.Duration) Interface {
+func NewBackupSyncController(
+	client arkv1client.BackupsGetter,
+	backupService cloudprovider.BackupService,
+	bucket string,
+	syncPeriod time.Duration,
+	logger *logrus.Logger,
+) Interface {
 	if syncPeriod < time.Minute {
-		glog.Infof("Backup sync period %v is too short. Setting to 1 minute", syncPeriod)
+		logger.WithField("syncPeriod", syncPeriod).Info("Provided backup sync period is too short. Setting to 1 minute")
 		syncPeriod = time.Minute
 	}
 	return &backupSyncController{
 		client:        client,
 		backupService: backupService,
 		bucket:        bucket,
 		syncPeriod:    syncPeriod,
+		logger:        logger,
 	}
 }
 
 // Run is a blocking function that continually runs the object storage -> Ark API
 // sync process according to the controller's syncPeriod. It will return when it
 // receives on the ctx.Done() channel.
 func (c *backupSyncController) Run(ctx context.Context, workers int) error {
-	glog.Info("Running backup sync controller")
+	c.logger.Info("Running backup sync controller")
 	wait.Until(c.run, c.syncPeriod, ctx.Done())
 	return nil
 }
 
 func (c *backupSyncController) run() {
-	glog.Info("Syncing backups from object storage")
+	c.logger.Info("Syncing backups from object storage")
 	backups, err := c.backupService.GetAllBackups(c.bucket)
 	if err != nil {
-		glog.Errorf("error listing backups: %v", err)
+		c.logger.WithField("error", err).Error("error listing backups")
 		return
```

Comment:
see above; it would be better if run could return an error rather than logging it into the ether.